### ISSUE TITLE
Regenerating hydrators do not refresh them

### ISSUE THREAD

**Issue by mirfilip** [Ocramius/GeneratedHydrator#29]
First of all, I'm not sure whether this is a bug, an expected behaviour or a missing feature, so please go easy :)

I try to use pre-generating hydrators feature and hook it up with my deployment system. I've created `generate.php` script with a snippet doing basically what documentation says (and it works fine). I, then, intend to use it with `pre-autoload-dump` composer event script in order **to achieve fresh hydrators** for my objects every time I run `composer install/update` (regardless if it's development or production env).

The reason I want **fresh** hydrators every time is that the default values / attributes can easily change during development. 

Because of a class existence check in `HydratorFactory::getHydratorClass()`, `HydratorFactory` will not attempt to create a proxy class when it already exists. I think pre-generation is intended for production usage only, when classes are not meant to get changed (thus hydrators don't need refreshing). Anyway, removing this class existence check makes it fit my scenario. How do you feel about `forceRefresh` kind of flag @Ocramius ?

Alternatively, do you have any ideas how to work it around ?

**Comment by Ocramius** [comment]
@mirfilip good point. I think the simplest way would be to implement signing like Ocramius/ProxyManager#172

**Comment by Ocramius** [comment]
@mirfilip consider that I don't have time to fix this bug right now, so consider sending a PR after looking in the linked PR.

**Comment by mirfilip** [comment]
@Ocramius I'll try to have a go, sure.

**Comment by gonzaloserrano** [comment]
Hey guys, we just hit the same issue. Any further news about this?

**Comment by mirfilip** [comment]
@gonzaloserrano, I'll be looking at this this week(end), stay in touch

**Comment by gonzaloserrano** [comment]
Hi again, i found the root cause of my issue and i think it's not related.

Somehow some (3) files from generated files stored in ```/tmp``` had root permissions, so then when the default generation strategy was executed the rename here https://github.com/Ocramius/CodeGenerationUtils/blob/master/src/CodeGenerationUtils/GeneratorStrategy/FileWriterGeneratorStrategy.php#L78 failed, which caused two problems:

- ```/tmp``` had thousands of generated temporal files
- the new files where not substituting the old ones because the ```rename()``` failed so they had the old structure

I have changed those file permissions and i think now it works as expected.

Thanks for your time.
